### PR TITLE
[cluster-test] Migrate debug_interface_log_tail to async api

### PR DESCRIPTION
**Pull request by changyonghu**
Closes: #1863
Referred #2844 
Manually tested transaction generator run and health check run in swarm mode and ./scripts/cti --pr <PR> --run bench in normal mode.

### CODE REVIEW

File: common/debug-interface/src/lib.rs [hunk 1]
```diff
@@ -59,3 +59,56 @@ impl NodeDebugClient {
...
+
+/// Implement default utility client for AsyncNodeDebugInterface
+pub struct AsyncNodeDebugClient {
+    client: reqwest::Client,
+    addr: String,
+}
+
+impl AsyncNodeDebugClient {
+    /// Create AsyncNodeDebugInterface from a valid socket address.
+    pub fn new<A: AsRef<str>>(address: A, port: u16) -> Self {
```
Comment (andll) [inline common/debug-interface/src/lib.rs:71]: Can we pass reqwest::Client here? It is a good practice to reuse same Client to save resources

Comment (changyonghu) [reply common/debug-interface/src/lib.rs:71]: I don't quite get it. Where is the same Client from? Like we check if there is any client for this IP address first?

Comment (andll) [reply common/debug-interface/src/lib.rs:71]: We just want to one Client instance and pass this (cloned) instance into each  AsyncNodeDebugClient , as an example you can look at how `Cluster::from_host_port` shares single Client across all Instance.

[As a note, cloning Client only clones 'handle', but resources are shared]

You don't need to check one client per IP or smth like that.

Comment (changyonghu) [reply common/debug-interface/src/lib.rs:71]: Can we reuse the http Client in Cluster for this purpose? I don't know what is the http client used for?

Comment (andll) [reply common/debug-interface/src/lib.rs:71]: Yes, we ca reuse http client from cluster, or create a new one, just don't want to create O(N) http clients

**Comment by andll** [comment]
Great job!

**Pull request closed without merging**